include/fmt/format.h

```diff
@@ -1871,22 +1879,75 @@ auto write_float(OutputIt out, const DecimalFP& fp,
   });
 }
 
+#ifdef __cpp_lib_bit_cast
+template <typename T, FMT_ENABLE_IF(is_fast_float<T>::value)>
+constexpr bool is_infinite_fast_float(T value) {
+  using floaty =
+      std::conditional_t<std::is_same<T, long double>::value, double, T>;
+  using uint = typename dragonbox::float_info<floaty>::carrier_uint;
+  constexpr auto significand_bits =
+      dragonbox::float_info<floaty>::significand_bits;
+  auto bits = std::bit_cast<uint>(value);
+  return (bits & exponent_mask<floaty>()) &&
+         !(bits & ((uint(1) << significand_bits) - 1));
+}
+
+template <typename T, FMT_ENABLE_IF(is_fast_float<T>::value)>
+constexpr bool is_finite_fast_float(T value) {
+  using floaty =
+      std::conditional_t<std::is_same<T, long double>::value, double, T>;
+  using uint = typename dragonbox::float_info<floaty>::carrier_uint;
+  auto bits = std::bit_cast<uint>(value);
+  return (bits & exponent_mask<floaty>()) != exponent_mask<floaty>();
+}
+#endif
+
+template <typename T, FMT_ENABLE_IF(std::is_floating_point<T>::value)>
+FMT_INLINE FMT_CONSTEXPR bool float_signbit(T value) {
+  if (is_constant_evaluated()) {
+#ifdef __cpp_if_constexpr
+    if constexpr (is_fast_float<T>::value) {
+      using floaty =
+          conditional_t<std::is_same<T, long double>::value, double, T>;
+      using uint = typename dragonbox::float_info<floaty>::carrier_uint;
+      auto bits = bit_cast<uint>(value);
+      return (bits & (uint(1) << (num_bits<uint>() - 1))) != 0;
+    } else {
+      FMT_ASSERT(false, "floating point type is not supported");
+    }
+#endif
+  }
```

Comment:
Since casts preserve sign, I suggest casting to `double`, checking `numeric_limits::is_iec559` and using `uint64_t` instead of dragonbox internals. This will make the code more future proof.

Reply:
Done.